content/200-orm/050-overview/500-databases/950-cloudflare-d1.mdx

```diff
@@ -0,0 +1,47 @@
+---
+title: 'Cloudflare D1'
+metaTitle: 'Cloudflare D1'
+metaDescription: 'Guide to Cloudflare D1'
+tocDepth: 3
+preview: true
+---
+
+<TopBlock>
+
+This guide discusses the concepts behind using Prisma ORM and Cloudflare D1, explains the commonalities and differences between Cloudflare D1 and other database providers, and leads you through the process for configuring your application to integrate with Cloudflare D1.
+
+Prisma ORM support for Cloudflare D1 is currently in [Preview](/orm/more/releases#preview). We would appreciate your feedback [on GitHub](tbd).
+
+</TopBlock>
+
+## What is Cloudflare D1?
+
+D1 is Cloudflare’s native serverless database. It’s based on SQLite and can be used when deploying applications with Cloudflare. D1 was initially [launched in 2022](https://blog.cloudflare.com/introducing-d1) and is currently in a public beta stage.
+
+Following Cloudflare’s principles of geographic distribution and bringing compute and data closer to application users, D1 supports automatic read-replication. It dynamically manages the number of database instances and locations of read-only replicas based on how many queries a database is getting, and from where.
+
+For write-operations, queries travel to a single primary instance in order to propagate the changes to all read-replicas and ensure data consistency.
+
+## Commonalities with other database providers
+
+D1 is based on SQLite.
+
+Many aspects of using Prisma ORM with D1 are just like using Prisma ORM with any other relational database. You can still:
+
+- model your database with the [Prisma Schema Language](/orm/prisma-schema)
+- use Prisma ORM's existing [`sqlite` database connector](/orm/overview/databases/sqlite) in your schema
+- use [Prisma Client](/orm/prisma-client) in your application to talk to the database server at D1
+
+## Differences to consider
+
+There are a number of differences between D1 and SQLite to consider. You should be aware of the following when deciding to use D1 and Prisma ORM:
+
+- **Local and remote D1 (SQLite) databases**. Cloudflare provides local and remote versions of D1. The [local](https://developers.cloudflare.com/d1/configuration/local-development) version is managed using the `--local` option of the `wrangler d1` CLI and is located in `.wranger/state`. The [remote](https://developers.cloudflare.com/d1/configuration/remote-development) version is managed by Cloudflare and is accessed via HTTP.
```

Comment:
> and is located in `.wranger/state`

I think that is an implementation detail, it should be enough that it is stored locally as a SQLite database (without mentioning the location).

Reply:
If so, then I would put this under a section that explains how to use Introspection and Migration tooling of Prisma with D1. That is the only context in which you should know about this, so we might potentially document it. Here it feels a bit out of place to me.

Reply:
do you think it hurts to include that here? an alternative version could be:

```markdown
- `--local`: Executes the statement against a _local_ version of D1. This local version of D1 is a SQLite database file that'll be located in your project. This approach is useful, when you want to develop and test your Worker on your local machine. Learn more in the [Cloudflare docs](https://developers.cloudflare.com/d1/configuration/local-development/).
```

that still mentions that the file is located in your project but doesn't provide the precise location. I feel like this version then just leaves the reader wondering: "ok, thanks for the info, but where exactly in my project?".

> That is the only context in which you should know about this, so we might potentially document it. 

disagree. for me it was useful because I found the file and openend in TablePlus to verify that the dummy record had been created later on. 

> If so, then I would put this under a section that explains how to use Introspection and Migration tooling of Prisma with D1. 

I'm with you that we should put that info there _as well_ (but not exclusively).